pkg/registry/flowcontrol/ensurer/flowschema_test.go

```diff
@@ -0,0 +1,350 @@
+			bootstrap:         newFlowSchema("fs1", "pl2", 200).Object(),
+			newObjectExpected: nil,
+			updateExpected:    false,
+		},
+		{
+			name:              "auto update is disabled, second generation, spec matches - no update expected",
+			current:           newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("false").WithGeneration(2).Object(),
+			bootstrap:         newFlowSchema("fs1", "pl1", 100).Object(),
+			newObjectExpected: nil,
+			updateExpected:    false,
+		},
+		{
+			name:              "annotation is missing, first generation, spec does not match - both annotation and spec update expected",
+			current:           newFlowSchema("fs1", "pl1", 100).WithGeneration(1).Object(),
+			bootstrap:         newFlowSchema("fs1", "pl2", 200).Object(),
+			newObjectExpected: newFlowSchema("fs1", "pl2", 200).WithAutoUpdateAnnotation("true").WithGeneration(1).Object(),
+			updateExpected:    true,
+		},
+		{
+			name:              "annotation is missing, first generation, spec matches - annotation update is expected",
+			current:           newFlowSchema("fs1", "pl1", 100).WithGeneration(1).Object(),
+			bootstrap:         newFlowSchema("fs1", "pl1", 100).Object(),
+			newObjectExpected: newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("true").WithGeneration(1).Object(),
+			updateExpected:    true,
+		},
+		{
+			name:              "annotation is missing, second generation, spec does not match - annotation update is expected",
+			current:           newFlowSchema("fs1", "pl1", 100).WithGeneration(2).Object(),
+			bootstrap:         newFlowSchema("fs1", "pl2", 200).Object(),
+			newObjectExpected: newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("false").WithGeneration(2).Object(),
+			updateExpected:    true,
+		},
+		{
+			name:              "annotation is missing, second generation, spec matches - annotation update is expected",
+			current:           newFlowSchema("fs1", "pl1", 100).WithGeneration(2).Object(),
+			bootstrap:         newFlowSchema("fs1", "pl1", 100).Object(),
+			newObjectExpected: newFlowSchema("fs1", "pl1", 100).WithAutoUpdateAnnotation("false").WithGeneration(2).Object(),
+			updateExpected:    true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			strategy := newSuggestedEnsureStrategy(&flowSchemaWrapper{}, false)
+			newObjectGot, updateGot, err := strategy.ShouldUpdate(test.current, test.bootstrap)
+			if err != nil {
+				t.Errorf("Expected no error, but got: %v", err)
+			}
+			if test.updateExpected != updateGot {
+				t.Fatalf("Expected update=%t but got: %t", test.updateExpected, updateGot)
+			}
+
+			if !test.updateExpected {
+				if newObjectGot != nil {
+					t.Errorf("Expected a nil object, but got: %#v", newObjectGot)
+				}
+				return
+			}
+
+			if !reflect.DeepEqual(test.newObjectExpected, newObjectGot) {
+				t.Errorf("Expected the object to be updated to match - diff: %s", cmp.Diff(test.newObjectExpected, newObjectGot))
+			}
+		})
+	}
+}
```

Comment:
Please add a test for Remover

Same for priority level.

Reply:
tests added for both